src/main/java/org/ebml/matroska/MatroskaFileWriter.java

```diff
@@ -246,6 +249,16 @@ public void addFrame(final MatroskaFileFrame frame)
     {
       cueData.addCue(ioDW.getFilePointer(), frame.getTimecode(), frame.getTrackNo());
     }
+
+    long defaultDuration = 0;
+
+    MatroskaFileTrack tr = tracks.getTrack(frame.getTrackNo());
+    if (tr != null)
+    {
+      defaultDuration = tr.getDefaultDuration();
+    }
+
+    lastDuration = cluster.getLastTimecode() + TimeUnit.NANOSECONDS.toMillis(defaultDuration);;
```

Comment:
This assumes that the first timecode is 0, and that frames from all tracks are always added in order (e.g. addFrame(track=1, timecode=1000); addFrame(track=2, timecode=900) is handled incorrectly and we may end up using something like that after buffering in jmr). 

I suggest just saving the smallest and largest timecode of all frames added to the segment, then setting duration to the difference. `cluster.getLastTimecode()` is always just `frame.getTimecode()`, so no need to expose it

Reply:
Changed that, thanks.